addons/l10n_de/migrations/2.0/pre-migrate.py

```diff
@@ -12,16 +12,25 @@ def rename_tag(cr, old_tag, new_tag):
 
 
 def migrate(cr, version):
-    rename_tag(cr, "tag_de_liabilities_bs_C_1", "tag_de_liabilities_bs_B_1")
-    rename_tag(cr, "tag_de_liabilities_bs_C_2", "tag_de_liabilities_bs_B_2")
-    rename_tag(cr, "tag_de_liabilities_bs_C_3", "tag_de_liabilities_bs_B_3")
-    rename_tag(cr, "tag_de_liabilities_bs_D_1", "tag_de_liabilities_bs_C_1")
-    rename_tag(cr, "tag_de_liabilities_bs_D_2", "tag_de_liabilities_bs_C_2")
-    rename_tag(cr, "tag_de_liabilities_bs_D_3", "tag_de_liabilities_bs_C_3")
-    rename_tag(cr, "tag_de_liabilities_bs_D_4", "tag_de_liabilities_bs_C_4")
-    rename_tag(cr, "tag_de_liabilities_bs_D_5", "tag_de_liabilities_bs_C_5")
-    rename_tag(cr, "tag_de_liabilities_bs_D_6", "tag_de_liabilities_bs_C_6")
-    rename_tag(cr, "tag_de_liabilities_bs_D_7", "tag_de_liabilities_bs_C_7")
-    rename_tag(cr, "tag_de_liabilities_bs_D_8", "tag_de_liabilities_bs_C_8")
-    rename_tag(cr, "tag_de_liabilities_bs_E", "tag_de_liabilities_bs_D")
-    rename_tag(cr, "tag_de_liabilities_bs_F", "tag_de_liabilities_bs_E")
+    # As some people already upgraded, they will have renamed the C_1 tag to B_1. This doesn't come from the script but from the
+    # account_account_tags_data.xml. If they try to run this script now in the fix they get the error that B_1 already exist.
+    # To fix this we can check if it exists or not, and if it does then we don't run the script. This means that the ones
+    # that upgraded won't have the old tags data transferred to the new tags but they will still be able to have the updated sheet.
+    cr.execute(
+        """SELECT 1 FROM ir_model_data
+           WHERE module='l10n_de' AND name='tag_de_liabilities_bs_B_1'
+        """)
+    if not cr.fetchone():
```

Comment:
This avoid indenting the whole file.
```suggestion
    cr.execute("SELECT 1 FROM ir_model_data  WHERE module='l10n_de' AND name='tag_de_liabilities_bs_B_1'")
    if cr.rowcount:
        return
```

Reply:
What if it's already used a FK in another table? Isn't it better to add it with the note that the tag is deprecated?

Reply:
> What if it's already used as a FK in another table?

Nothing will happen as the ORM won't do anything. That's more or less the basis of `util.delete_unused`. Note the entry in ir_model_data is not directly tied to the entries in the model tables, having a FK to ir_model_data is highly unlikely and probably a mistake anyway.

@KangOl wdyt? I think we could also set `noupdate=True` instead of removing from `ir_model_data`

Reply:
I think I'm missing something, can you explain a bit more? When we delete it from the ir_model_data table, it won't be deleted from the tags table? Because I think the FK is to the tags table not the ir_model_data

Reply:
Yes. Basically the ORM starts and looks in ir_model_data, and roughly:
1. any extra xmlid mean something to remove,
2. any new xlmid means something to create
3. existing xmlids matching what it knows about means something to update.

This is affected by the nopupdate flags.

Thus, regarging my suggestion, if you remove an entry from ir_model_data when the ORM starts it won't remove the linked record because it doesn't have the info. For the ORM it would be like if the record was created by the client (not coming from data).

Reply:
Note: that's the reason of `pre-` scripts -> to change stuff _brefore_ the ORM tries to create/remove/update data itself ;)